ext/opcache/Optimizer/zend_cfg.h

```diff
@@ -42,17 +42,19 @@
 #define ZEND_BB_PROTECTED        (ZEND_BB_ENTRY|ZEND_BB_RECV_ENTRY|ZEND_BB_TRY|ZEND_BB_CATCH|ZEND_BB_FINALLY|ZEND_BB_FINALLY_END|ZEND_BB_GEN_VAR|ZEND_BB_KILL_VAR)
 
 typedef struct _zend_basic_block {
+	int              *successors;         /* successor block indices     */
 	uint32_t          flags;
 	uint32_t          start;              /* first opcode number         */
 	uint32_t          len;                /* number of opcodes           */
-	int               successors[2];      /* up to 2 successor blocks    */
+	int               successors_count;   /* number of successors        */
 	int               predecessors_count; /* number of predecessors      */
 	int               predecessor_offset; /* offset of 1-st predecessor  */
 	int               idom;               /* immediate dominator block   */
 	int               loop_header;        /* closest loop header, or -1  */
 	int               level;              /* steps away from the entry in the dom. tree */
 	int               children;           /* list of dominated blocks    */
 	int               next_child;         /* next dominated block        */
+	int               successors_storage[2]; /* up to 2 successor blocks */
```

Comment:
I understand your point, and I have to agree, despite, I afraid of big changes in optimizer core code.